### DISCUSSION TITLE
Slot for original filename in las object

### DISCUSSION THREAD

**Discussion by azh2** [r-lidar/lidR#453] [Ideas]
I came across a scenario where it might be useful to have a slot containing the original filename for an las object, similarly to the @file@name slot in a raster object. Specifically, I am trying to cross-reference an las object in a user-defined catalog function to read in a file containing the same ID in the filename.

**Comment by Jean-Romain** [comment]
I understand the request and I may have needed a similar feature myself. 

The problem is that the filename of a `LAS` object is not always well-defined. Actually the only case it is well defined is when processing a `LAScatalog` by file. This is likely the most common use case but is actually a special case of the `LAScatalog` processing engine. When processing smaller or larger chunks it is irrelevant because one `LAS` object may be made of parts from many LAS files. Thus, the filename is non-definable. Similarly when using `readLAS()` you can read multiple file as one `LAS`. Same problem.

This is why what you are asking is not in `lidR` and won't be in `lidR`. However I'd like to know what is the scenario you are working with. Maybe we can find something better and more robust that could be used to improve the package.
